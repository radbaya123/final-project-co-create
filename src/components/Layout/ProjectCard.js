import React from "react";
import { Descriptions, Button, Card, Space } from "antd";

const Item = Descriptions.Item;

function ProjectCard(props) {
  const { category, name, date, description, collaborator, link, invited_user_id } = props;
  return (
    <Card style={{ marginBottom: "1.5rem" }}>
      <Descriptions
        bordered
        size="small"
        extra={
          <Space>
            <Button type="primary" /*onClick={props.btn_click}*/>
              Edit
            </Button>
            <Button type="danger" onClick={props.btn_click}>
              Hapus
            </Button>
          </Space>
        }
      >
        <Descriptions.Item label="Kategori Project" span={24}>
          {category}
        </Descriptions.Item>
        <Descriptions.Item span={24} label="Nama Project">
          {name}
        </Descriptions.Item>
        <Descriptions.Item span={24} label="Tanggal Mulai">
          {date}
        </Descriptions.Item>
        <Descriptions.Item span={24} label="Deskripsi Project">
          {description}
        </Descriptions.Item>
        <Descriptions.Item span={24} label="Link Trello">
          <a href={link} target="_blank" rel="noreferrer">
            {link}
          </a>
        </Descriptions.Item>
        <Descriptions.Item span={24} label="Member yang Kamu Undang">
          {invited_user_id}
        </Descriptions.Item>
        <Descriptions.Item span={24} label="Kolaborator">
          {collaborator}
        </Descriptions.Item>
      </Descriptions>
    </Card>
  );
}

export default ProjectCard;
