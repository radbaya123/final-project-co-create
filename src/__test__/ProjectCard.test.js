// Dependencies
import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

// Components
import ProjectCard from "components/Layout/ProjectCard";
// import MyProject from "components/Layout/MyProject";/
import { Descriptions } from "antd";

const { Item } = Descriptions;

//Configure enzyme for React 17
Enzyme.configure({ adapter: new Adapter() });

// test("test your component", () => {
// 	const wrapper = mount(<App2 txt={"michaelangelo"} />);

// 	expect(wrapper.find("p").text()).toEqual("hi michaelangelo");
// });

describe(`Test ProjectCard component`, () => {
	const wrapper = shallow(<ProjectCard category="Game" />);

	it(`Should render MyProjectCard`, () => {
		expect(wrapper).toMatchSnapshot();
	});
	it(`Should render category props`, () => {
		expect(wrapper.find({ "data-testid": "category" })).toEqual({});
		// expect(wrapper.find({ "data-testid": "category" }).length).toBe(1);
	});
});