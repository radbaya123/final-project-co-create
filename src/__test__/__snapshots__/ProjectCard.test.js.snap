// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProjectCard component Should render MyProjectCard 1`] = `ShallowWrapper {}`;
